pubsub/google/cloud/pubsub_v1/subscriber/_protocol/streaming_pull_manager.py

```diff
@@ -116,6 +117,12 @@ def __init__(
         else:
             self._scheduler = scheduler
 
+        # A FIFO queue for the messages that have been received from the server,
+        # but not yet added to the lease management (and not sent to user callback),
+        # because the FlowControl limits have been hit.
+        self._messages_on_hold = queue.Queue()
```

Comment:
Do we need to set a maximum size on the queue?

Reply:
We don't - the queue only gets filled in the `_on_response()` callback, and only after the load reaches 1.0. At that point the consumer is already paused and no more messages arrive.

Additionally, the consumer only gets maybe-resumed after clearing the queue first, meaning that the maximum queue size is bounded by the maximum number of messages the server sends in a single response.